packages/web-components/src/components/masthead/masthead-logo.ts

```diff
@@ -35,15 +51,66 @@ class DDSMastheadLogo extends FocusMixin(LitElement) {
   @property({ reflect: true })
   slot = 'brand';
 
+  /**
+   * Search bar opened flag.
+   */
+  @internalProperty()
+  private _openedSearch = false;
+
+  /**
+   * Large breakpoint flag.
+   */
+  @internalProperty()
+  private _responsiveLg = false;
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  @HostListener('parentRoot:eventToggle')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleSearchToggle = (event: Event) => {
+    this._openedSearch = (event as CustomEvent).detail.active;
+  };
+
+  /**
+   * Handles media query event.
+   *
+   * @param event The event.
+   */
+  private _handleMediaQuery(event: Event) {
+    this._responsiveLg = (event as MediaQueryListEvent).matches;
+  }
+
   createRenderRoot() {
     return this.attachShadow({ mode: 'open', delegatesFocus: true });
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    const mqLarge = this.ownerDocument?.defaultView?.matchMedia(`(max-width: ${breakpoints.lg.width})`);
+    this._responsiveLg = mqLarge ? mqLarge.matches : false;
+
+    this._mqHandler = on(mqLarge, 'change', this._handleMediaQuery.bind(this));
+  }
+
+  disconnectedCallback() {
+    if (this._mqHandler) {
+      this._mqHandler.release();
```

Comment:
```suggestion
      this._mqHandler = this._mqHandler.release();
```

This codebase nulls out handle properties when they are no longer in use.